lib/public/utilities/fetch/RemoteDataSource.js

```diff
@@ -10,35 +10,39 @@
  *  granted to it by virtue of its status as an Intergovernmental Organization
  *  or submit itself to any jurisdiction.
  */
-import { Observable, RemoteData } from '/js/src/index.js';
+import { RemoteData } from '/js/src/index.js';
 import { getRemoteData } from './getRemoteData.js';
 import { ObservableData } from '../ObservableData.js';
+import { DataSource } from '../DataSource.js';
 
 /**
- * Service class providing observable data fetching process
+ * Class to abstract notifying from data changes
+ * @template T - type of response sent by server
  */
-export class RemoteDataSource extends Observable {
+export class RemoteDataSource extends DataSource {
     /**
      * Constructor
      */
     constructor() {
         super();
 
+        this._default = () => RemoteData.notAsked();
+
         /**
-         * @type {ObservableData<RemoteData>}
+         * @type {ObservableData<RemoteData<T>}
          * @private
          */
-        this._observableData = new ObservableData(RemoteData.notAsked());
+        this._observableData = new ObservableData(this._default());
         this._abortController = null;
     }
 
     /**
      * Fetch the given endpoint to fill current data
      *
-     * @param {string} endpoint the endpoint to fetch
+     * @param {string} [args.endpoint] the endpoint to fetch
      * @return {Promise<void>} resolves once the data fetching has ended
      */
-    async fetch(endpoint) {
+    async fetch({ endpoint }) {
```

Comment:
I think that here endpoint should be a parameter on its own, it will never be omited